pkgs/applications/editors/oni2/default.nix

```diff
@@ -0,0 +1,273 @@
+          cp $3/package{-upd,}.json
+
+          # Copy custom yarn.lock to match updated package.json, do fixup
+          cp $2 $3/yarn.lock
+          fixup_yarn_lock $3/yarn.lock
+
+          # Make yarn install prefetched dependencies
+          yarn config --offline set yarn-offline-mirror $1
+          # Set explicit node install directory for node-gyp.
+          npm_config_nodedir=${nodejs} yarn install --frozen-lockfile --offline --no-progress --non-interactive --cwd $3
+        }
+        yarnInstall ${node} ${./node.lock} node
+        yarnInstall ${extensions} ${./extensions.lock} extensions
+
+        export ESY__PREFIX="$NIX_BUILD_TOP/esy"
+        esy '@release' install # should do nothing
+
+        export ESY__GLOBAL_PATH=PATH
+        # Create link to bin directory, currently empty
+        esy '@release' sh -c "ln -s \$cur__bin result"
+        # Finish building Oni2
+        esy '@release' x Oni2 --help
+
+        runHook postBuild
+      '';
+
+      installPhase = ''
+        runHook preInstall
+
+        mkdir $out
+
+        cp -Lr ./result $out/bin
+        cp -r ./node $out/
+        cp -r ./extensions $out/
+
+        chmod +w $out/bin
+        chmod +x $out/bin/Oni2 $out/bin/Oni2_editor
+        # Unset LANG. See https://github.com/onivim/oni2/issues/3772
+        mv $out/bin/Oni2{,_unwrapped}
+        makeWrapper $out/bin/Oni2{_unwrapped,} --unset LANG
+        mv $out/bin/Oni2_editor{,_unwrapped}
+        makeWrapper $out/bin/Oni2_editor{_unwrapped,} --unset LANG
+
+        rm -f $out/bin/setup.json
+        jq -n "{node: \"${nodejs}/bin/node\", nodeScript: \"$out/node\", bundledExtensions: \"$out/extensions\", rg: \"${ripgrep}/bin/rg\"}" > $out/bin/setup.json
+
+        mkdir -p $out/share/applications $out/share/pixmaps
+        cp ${source}/scripts/linux/Onivim2.desktop $out/share/applications
+        cp ${source}/assets/images/icon512.png $out/share/pixmaps/Onivim2.png
+
+        runHook postInstall
+      '';
+
+      meta = with lib; {
+        description = "Native, lightweight modal code editor";
+        longDescription = ''
+          Onivim 2 is a reimagination of the Oni editor. Onivim 2 aims to bring the speed of Sublime, the language integration of VSCode, and the modal editing experience of Vim together, in a single package.
+        '';
+        homepage = "https://v2.onivim.io/";
+        inherit license;
+        maintainers = with maintainers; [ gardspirito ];
+        platforms =
+          [ "x86_64-linux" "x86_64-darwin" ];
+      };
+    };
+in mkOni2 {
+  variant = "oni2";
+  license = {
+    fullName = "Outrun Labs End User License Agreement";
+    url =
+      "https://github.com/onivim/oni2/blob/master/Outrun-Labs-EULA-v1.1.md";
+    free = false;
+  };
+  version = "0.5.7";
+  rev = "v0.5.7";
+  sha256 = "NlN0Ntdwtx5XLjd1ltUzv/bjmJQR5eyRqtmicppP6YU=";
+  fetchDepsSha256 = "k7G6jPJfxCCSuSucPfiXljCVJhmjl/BxWMCEjv2tfhA=";
```

Comment:
Got hash mismatch.
Perhaps this should be `qxQ7Dznxqv6yqMaXyWqkCN8cwlU/BY8jL3HHF486OEw=`?
When in doubt, try changing 1 char of the hash in your machine and see if it spits the hash above.

Reply:
That's odd. I did recheck — I always get hash `k7G6jPJfxCCSuSucPfiXljCVJhmjl/BxWMCEjv2tfhA=`. Can you please send me your realization of `oni2-fetch-deps` so I can make a comparison and understand the source of the problem?

And yeah, what is your OS?

Reply:
Sure.
NixOS 21.11.20210828.9bd51f0 (Porcupine) x86_64

didn't got that error when I ran `nix-build -A oni2`, so I guess It's something with `nixpkgs-review`.

By the way, are you sure that you have checked the execution of the binaries? ran `./result/bin/Oni2` and nothing happened. no errors, program just closes instantly.

Reply:
Rechecked with `nixpkgs-review` on latest revision of nixpkgs. I had no problems.

About execution of the binaries: I checked execution of them on my machines and in the GNOME VM, so this may be related to bugs in Oni2 itself. I would be truly grateful if you:
- Checked for presence of `onivim2-crash.log` in your directory. Usually, Oni2 writes there in case of crash.
- Attempted to run binaries in `nix-shell --pure -p`. Some environmental variables you currently have may cause crash. I know for sure that unsupported `LANG` causes silent crash, so I had to manually unset it through wrapper.
 - If binaries work fine in pure environment, can you please dump a censored version of your environment variables?

Reply:
running it in Pure environment works.
Dumping vars now: https://hastebin.com/nopogesuje.env